browser/hub/internal/android/java/java_sources.gni

```diff
@@ -9,3 +9,5 @@ brave_browser_hub_internal_java_sources = [
   "//brave/browser/hub/internal/android/java/src/org/chromium/chrome/browser/hub/BraveHubManagerImpl.java",
   "//brave/browser/hub/internal/android/java/src/org/chromium/chrome/browser/hub/BraveHubToolbarView.java",
 ]
+
+brave_browser_hub_internal_java_deps = [ "//chrome/browser/preferences:java" ]
```

Comment:
Can we create BUILD.gn instead of sources.gni here ?

Reply:
I don't understand what you mean @deeppandya, this is being added as a dep, but I'm not understanding why we need to add an upstream dep to an upstream target?

Reply:
some comments would be helpful here to explain

Reply:
i meant adding BUILD.gn for the package similar to upsteram instead of sources. but i think it's out of the scope for the PR changes.

Reply:
@deeppandya 
the deps I have added is required to provide ability do `import org.chromium.chrome.browser.preferences.ChromeSharedPreferences;`
at https://github.com/brave/brave-core/pull/28679/commits/b59f6dd2a5b01a5f92590562e802b47611f92f9a#diff-ee2bfd2441d8c464a8c54ce22d3a082e13dbf3e9e05e91c9ea08458c280d1588R19

and the `class BraveHubManagerImpl` is involved into bytecode patching at
https://github.com/brave/brave-core/blob/master/build/android/bytecode/java/org/brave/bytecode/BraveHubManagerImplClassAdapter.java#L18

so I don't know, how will it work.

I don't remember, do we have already examples of BraveClass.java and ChromeClass.java which would be placed in different targets, but worked god with bytecode?

Reply:
@deeppandya BUILD.gn for which package? 
@AlexeyBarabash I'm not saying this is wrong necessarily, but it struck me as unusual to add upstream deps to an existing upstream target so some comments would be helpful here

Reply:
Oh, I didn't realize that file existed, this added dep should go in that same file and then it's fine

Reply:
also now I see what @deeppandya means about using a BUILD.gn there instead of sources.gni. It should be its own target and then remove `sources += brave_browser_hub_internal_java_sources`

Then `brave_browser_hub_internal_java_deps = [ "//brave/browser/hub/internal/android/java/src/org/chromium/chrome/browser/hub"]`

Reply:
don't use sources.gni unless there are circular dependencies with the target you are adding it to

Reply:
@bridiver 

There are circular dependencies.

I have tried the proposed approach here
https://github.com/brave/brave-core/pull/28679/commits/a701223325910d62ded3db8772b6d9bed11dd7fd

Got circular dependencies error, because:
1) `class BraveHubManagerImpl extends HubManagerImpl` => we must add 
```
deps += [ "//chrome/browser/hub/internal:java" ]
```
into new created `src/brave/browser/hub/internal/BUILD.gn:java`

2) when `BraveHubManagerImpl` sits in its own `BUILD.gn`, we add 
```
deps += brave_browser_hub_internal_java_deps # [ "//brave/browser/hub/internal:java" ]
```
into `src/chrome/browser/hub/internal/BUILD.gn:java`


Tried to use `allow_circular_includes_from` - found it works for *.h/*.cc files.

Reply:
here it is how does it look on CI
https://ci.brave.com/job/brave-core-build-pr-android/job/PR-28679/3/console
```
14:54:25  /home/ubuntu/workspace/e-core-build-pr-android_PR-28679/src
14:54:25  > gn gen /home/ubuntu/workspace/e-core-build-pr-android_PR-28679/src/out/android_Component --check
14:54:37  ERROR Dependency cycle:
14:54:37    //brave/browser/hub/internal:java__build_config_crbug_908819 ->
14:54:37    //chrome/browser/hub/internal:java__build_config_crbug_908819 ->
14:54:37    //brave/browser/hub/internal:java__build_config_crbug_908819
```